packages/ui/src/components/InputFields/PasswordInputField.tsx

```diff
@@ -0,0 +1,46 @@
+import type { ForwardedRef } from 'react';
+import { forwardRef, useRef } from 'react';
+import { useUpdateEffect, useToggle } from 'usehooks-ts';
+
+import PasswordHideIcon from '@/assets/icons/password-hide-icon.svg';
+import PasswordShowIcon from '@/assets/icons/password-show-icon.svg';
+import IconButton from '@/components/Button/IconButton';
+import useMergeRef from '@/hooks/use-merge-ref';
+
+import type { Props as InputFieldProps } from './InputField';
+import InputField from './InputField';
+
+type Props = Omit<InputFieldProps, 'type' | 'suffix' | 'isSuffixFocusVisible'>;
+
+const PasswordInputField = (props: Props, forwardRef: ForwardedRef<HTMLInputElement>) => {
+  const [showPassword, toggleShowPassword] = useToggle(false);
+  const innerRef = useRef<HTMLInputElement>(null);
+
+  // Merge inner ref with forwarded ref
+  const mergeRef = useMergeRef(forwardRef, innerRef);
```

Comment:
Use `useImperativeHandle` instead?
https://beta.reactjs.org/reference/react/useImperativeHandle

Reply:
it will overwrite the original forwardRef instance.

Reply:
This looks weird. Any particular reason?

Reply:
Updated per offline discussion